### ISSUE TITLE
Broken tests with Mono 4

### ISSUE THREAD

**Issue by rapgro** [icsharpcode/SharpZipLib#90]
Not sure if I should open several issues individually for each failed TestFixture.
```
+ nunit-console26 -labels '-exclude:Long Running' bin/debug/tests.dll
NUnit-Console version 2.6.4.0
Copyright (C) 2002-2012 Charlie Poole.
Copyright (C) 2002-2004 James W. Newkirk, Michael C. Two, Alexei A. Vorontsov.
Copyright (C) 2000-2002 Philip Craig.
All Rights Reserved.

Runtime Environment - 
   OS Version: Unix 4.2.1.300
  CLR Version: 4.0.30319.17020 ( Mono 4.0 ( 4.0.4 (Stable 4.0.4.1/5ab4c0d Mon Sep 14 11:53:28 UTC 2015) ) )

ProcessModel: Default    DomainUsage: Single
Execution Runtime: mono-4.0
Excluded categories: not Long Running
{…}

Errors and Failures:
1) Test Failure : ICSharpCode.SharpZipLib.Tests.GZip.GZipTestSuite.ZeroLengthInputStream
     reading from an empty stream should cause an exception
  Expected: True
  But was:  False

at ICSharpCode.SharpZipLib.Tests.GZip.GZipTestSuite.ZeroLengthInputStream () [0x00000] in <filename unknown>:0

2) Test Failure : ICSharpCode.SharpZipLib.Tests.Tar.TarTestSuite.LongNames
     Name match failure
  Expected string length 513 but was 512. Strings differ at index 512.
  Expected: "...                                                           A "
  But was:  "...                                                           A"
  --------------------------------------------------------------------------^

at ICSharpCode.SharpZipLib.Tests.Tar.TarTestSuite.TryLongName (System.String name) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Tests.Tar.TarTestSuite.LongNames () [0x00000] in <filename unknown>:0

3) Test Error : ICSharpCode.SharpZipLib.Tests.Zip.FastZipHandling.ReadingOfLockedDataFiles
   System.IO.IOException : Sharing violation on path /home/builder/rpmbuild/BUILD/SharpZipLib-e01215507cf25a5978a0bd850c9e67dbabf515b7/tests/bin/debug/a.dat
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean anonymous, FileOptions options) [0x00000] in <filename unknown>:0 
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.Open (System.String path, FileMode mode, FileAccess access, FileShare share) [0x00000] in <filename unknown>:0 
  at ICSharpCode.SharpZipLib.Zip.FastZip.ProcessFile (System.Object sender, ICSharpCode.SharpZipLib.Core.ScanEventArgs e) [0x00000] in <filename unknown>:0 

4) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.GeneralHandling.InvalidPasswordNonSeekable
   An unexpected exception type was thrown
Expected: ICSharpCode.SharpZipLib.Zip.ZipException
 but was: ICSharpCode.SharpZipLib.SharpZipBaseException : Unknown block type 7
at ICSharpCode.SharpZipLib.Zip.Compression.Inflater.Decode () [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Zip.Compression.Inflater.Inflate (System.Byte[] buffer, Int32 offset, Int32 count) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Zip.Compression.Streams.InflaterInputStream.Read (System.Byte[] buffer, Int32 offset, Int32 count) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Zip.ZipInputStream.BodyRead (System.Byte[] buffer, Int32 offset, Int32 count) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Zip.ZipInputStream.InitialRead (System.Byte[] destination, Int32 offset, Int32 count) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Zip.ZipInputStream.Read (System.Byte[] buffer, Int32 offset, Int32 count) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Tests.Zip.GeneralHandling.InvalidPasswordNonSeekable () [0x00000] in <filename unknown>:0

5) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.BasicDirectories
     Should be equal
  String lengths are both 14. Strings differ at index 8.
  Expected: "absolute\\file2"
  But was:  "absolute/file2"
  -------------------^

at ICSharpCode.SharpZipLib.Tests.Zip.TransformBase.TestDirectory (INameTransform t, System.String original, System.String expected) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.BasicDirectories () [0x00000] in <filename unknown>:0

6) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.BasicFiles
     Should be equal
  Expected string length 6 but was 5. Strings differ at index 4.
  Expected: "base/t"
  But was:  "baset"
  ---------------^

at ICSharpCode.SharpZipLib.Tests.Zip.TransformBase.TestFile (INameTransform t, System.String original, System.String expected) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.BasicFiles () [0x00000] in <filename unknown>:0

7) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.Replacement
     Should be equal
  Expected string length 3 but was 4. Strings differ at index 1.
  Expected: "c/_"
  But was:  "c\\/>"
  ------------^

at ICSharpCode.SharpZipLib.Tests.Zip.TransformBase.TestFile (INameTransform t, System.String original, System.String expected) [0x00000] in <filename unknown>:0
at ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.Replacement () [0x00000] in <filename unknown>:0

8) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.ReplacementChecking
   Expected an exception
at ICSharpCode.SharpZipLib.Tests.Zip.WindowsNameTransformHandling.ReplacementChecking () [0x00000] in <filename unknown>:0

9) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.ZipEntryFactoryHandling.CreatedValues
     Create time failure
  Expected: 2100-02-27 11:07:56.000
  But was:  2015-10-09 21:52:02.000

at ICSharpCode.SharpZipLib.Tests.Zip.ZipEntryFactoryHandling.CreatedValues () [0x00000] in <filename unknown>:0

10) Test Failure : ICSharpCode.SharpZipLib.Tests.Zip.ZipNameTransformHandling.FilenameCleaning
     Expected: 0
  But was:  1

at ICSharpCode.SharpZipLib.Tests.Zip.ZipNameTransformHandling.FilenameCleaning () [0x00000] in <filename unknown>:0
```

**Comment by rapgro** [comment]
Any news here? The above mentioned tests still fail with Mono 4.4.1
`opyright (C) 2002-2012 Charlie Poole.
Copyright (C) 2002-2004 James W. Newkirk, Michael C. Two, Alexei A. Vorontsov.
Copyright (C) 2000-2002 Philip Craig.
All Rights Reserved.

Runtime Environment - 
   OS Version: Unix 4.5.7.300
  CLR Version: 4.0.30319.42000 ( Mono 4.0 ( 4.4.1 (Nightly 4.4.1.0/4747417 Wed Jun 22 18:59:52 UTC 2016) ) )
`

**Comment by McNeight** [comment]
@rapgro 
Some of these tests have been fixed by #127 and the rest are Windows specific. For a list of the tests that I'm using with NUint3 on Travis for testing with Mono, look at ICSharpCode.SharpZipLib.Tests/PassingTests.txt
For a list of the tests that I'm using with NUnit3 on AppVeyor for testing with .NET, look at ICSharpCode.SharpZipLib.Tests\PassingTestsWindows.txt

After the 1.0 release, I'll be taking a much closer look at test coverage and making the tests more consistent.

-Neil
